pkg/proxy/ipsock_posix.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
```

Comment:
This is just pulling in some 3rd party go code so we can use it?

Reply:
There is no public golang API to create sockaddr_in structures for `syscall.Bind()` so I've pulled in the golang code to do this. The golang high level APIs do not allow to set socket options prior to `connect()` and `listen()` so I had to use the system calls directly.

Reply:
OK works for me.